src/dune_rules/cxx_rules.ml

```diff
@@ -0,0 +1,46 @@
+open! Stdune
+open! Dune_engine
+
+let header_file_content =
+  {|
+#if defined( __clang__ )
+  #define CCOMP clang
+#elif defined( __MINGW32__) || defined( __MINGW64__ )
+  #define CCOMP mingw
```

Comment:
This distinction isn't used anywhere (I think). Two options:

- mingw-w64 defines `__GNUC__`, so this could be omitted
- I haven't used/tested it, but it's possible (and done) to build mingw-w64 using clang - so if this distinction is to be kept, I think `__MINGW32__` should be tested _before_ `__clang__` since otherwise `__MINGW32__` trumps `__GNUC__` but not `__clang__`